lib/src/value/calculation.dart

```diff
@@ -158,6 +343,204 @@ class SassCalculation extends Value {
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value atan2(Object y, Object? x) {
+    y = _simplify(y);
+    x = x.andThen(_simplify);
+    var args = [y, if (x != null) x];
+    _verifyLength(args, 2);
+    _verifyCompatibleNumbers(args);
+    if (y is! SassNumber ||
+        x is! SassNumber ||
+        y.hasUnit('%') ||
+        x.hasUnit('%') ||
+        !y.hasCompatibleUnits(x)) {
+      return SassCalculation._("atan2", args);
+    }
+    return number_lib.atan2(y, x);
+  }
+
+  /// Creates a `rem()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value rem(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    _verifyCompatibleNumbers(args);
+    if (dividend is! SassNumber ||
+        modulus is! SassNumber ||
+        !dividend.hasCompatibleUnits(modulus)) {
+      return SassCalculation._("rem", args);
+    }
+    var result = dividend.modulo(modulus);
+    if ((modulus.value.signIncludingZero != dividend.value.signIncludingZero)) {
+      if (modulus.value.isInfinite) return dividend;
+      if (result == SassNumber(0)) {
+        return result.unaryMinus();
+      }
+      return result.minus(dividend);
+    }
+    return result;
+  }
+
+  /// Creates a `mod()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    _verifyCompatibleNumbers(args);
+    if ((dividend is! SassNumber || modulus is! SassNumber) ||
```

Comment:
Unnecessary parentheses:

```suggestion
    if (dividend is! SassNumber || modulus is! SassNumber ||
```